std-lib/list/list.lurk

```diff
@@ -38,6 +38,15 @@
     (length (lambda (lst) (length-iter lst 0))))
     (current-env))
 
+;; apply : a preliminary version of apply
+(letrec 
+    ((apply (lambda (f list)
+        (if (car list)
```

Comment:
Why are you checking to see if the car of the list is NIL? NIL shouldn't be treated specially when treated as an argument to `f`.

Reply:
I think that is intended as a check to see if the list is empty. I don't know if `(if list ...)` is falsey when the list is empty, but that would be nice.

What you're getting at though is that `nil` is a valid list item, right?

Reply:
Yes, that's right.

To your other point, NIL is the empty list, and in Lurk as in Common Lisp, NIL is the false value.

Reply:
In general, it only makes sense to be checking to see if a *list* is NIL (when traversing lists). If you're checking the CAR, that's probably wrong. Even if traversing a binary tree, you probably put the NIL check at the top of the function.

This is just a heuristic I'm noting since @durlicc is new to Lisp.